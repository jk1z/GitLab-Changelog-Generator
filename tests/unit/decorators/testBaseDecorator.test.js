const BaseDecorator = require("../../../app/decorators/base");
describe("Base decorator", () => {
    describe("#populateContentBucketByContents", () => {
        test("should populate content buckets with default label configs", () => {
            const baseDecorator = new BaseDecorator({
                labelConfigs: [
                    { name: "issues", title: "Closed issues", default: true },
                    { name: "mergeRequests", title: "Merged merge requests", default: true }
                ]
            });
            baseDecorator.populateContentBucketByContents([
                { message: "mock issue message", labels: [], defaultLabel: "issues" },
                { message: "mock MR message", labels: [], defaultLabel: "mergeRequests" }
            ]);
            expect(baseDecorator.labelBucket).toMatchSnapshot();
        });
        test("should populate content buckets with custom label configs", () => {
            const baseDecorator = new BaseDecorator({
                labelConfigs: [
                    { name: "issues", title: "Closed issues", default: true },
                    { name: "mergeRequests", title: "Merged merge requests", default: true },
                    { name: "features", title: "Features" },
                    { name: "fixes", title: "Bug Fixes" }
                ]
            });
            baseDecorator.populateContentBucketByContents([
                { message: "mock feature message", labels: ["features"], defaultLabel: "issues" },
                { message: "mock fixes message", labels: ["fixes"], defaultLabel: "mergeRequests" }
            ]);
            expect(baseDecorator.labelBucket).toMatchSnapshot();
        });
        test("should populate content buckets with mixed label configs", () => {
            const baseDecorator = new BaseDecorator({
                labelConfigs: [
                    { name: "issues", title: "Closed issues", default: true },
                    { name: "mergeRequests", title: "Merged merge requests", default: true },
                    { name: "features", title: "Features" },
                    { name: "fixes", title: "Bug Fixes" }
                ]
            });
            baseDecorator.populateContentBucketByContents([
                { message: "mock normal MR message", labels: [], defaultLabel: "mergeRequests" },
                { message: "mock normal issue message", labels: [], defaultLabel: "issues" },
                { message: "mock feature message", labels: ["features"], defaultLabel: "issues" },
                { message: "mock fixes message", labels: ["fixes"], defaultLabel: "mergeRequests" }
            ]);
            expect(baseDecorator.labelBucket).toMatchSnapshot();
        });
    });
});
