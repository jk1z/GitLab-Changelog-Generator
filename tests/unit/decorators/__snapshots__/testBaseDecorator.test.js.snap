// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base decorator #populateContentBucketByContents should populate content buckets with custom label configs 1`] = `
Object {
  "features": Array [
    "mock feature message",
  ],
  "fixes": Array [
    "mock fixes message",
  ],
  "issues": Array [],
  "mergeRequests": Array [],
}
`;

exports[`Base decorator #populateContentBucketByContents should populate content buckets with default label configs 1`] = `
Object {
  "issues": Array [
    "mock issue message",
  ],
  "mergeRequests": Array [
    "mock MR message",
  ],
}
`;

exports[`Base decorator #populateContentBucketByContents should populate content buckets with mixed label configs 1`] = `
Object {
  "features": Array [
    "mock feature message",
  ],
  "fixes": Array [
    "mock fixes message",
  ],
  "issues": Array [
    "mock normal issue message",
  ],
  "mergeRequests": Array [
    "mock normal MR message",
  ],
}
`;
